src/kOS/Utilities/VesselUtils.cs

```diff
@@ -318,7 +389,312 @@ public static object GetSolarPanelStatus(Vessel vessel)
 
         public static void SolarPanelCtrl(Vessel vessel, bool state)
         {
-            vessel.rootPart.SendEvent(state ? "Extend" : "Retract");
+            foreach (var p in vessel.parts)
+            {
+                foreach (var c in p.FindModulesImplementing<ModuleDeployableSolarPanel>())
+                {
+                    if (state) { c.Extend(); }else { c.Retract(); }
+                }
+            }
+        }
+
+        public static object GetRadiatorStatus(Vessel vessel)
+        {
+            var atLeastOneRadiator = false; // No radiators at all? Always return false
+
+            foreach (var p in vessel.parts)
+            {
+                foreach (var c in p.FindModulesImplementing<ModuleActiveRadiator>())
+                {
+                    atLeastOneRadiator = true;
+
+                    if (! c.IsCooling)
```

Comment:
Depends on what we want with parts that don't want to be deployed...
But in case of radiators this code references both deployable (which won't deploy and work if stowed) and fixed (which don't seem to radiate when stowed, but still drain heat from adjacent parts if active - acting as active heat sinks) radiators